refactor configuration

Spookerton opened the pull request:
makes config members static, which may speed some bits and pieces up somewhere
pulls old load_configuration inside & refactors config to be a little more nice to read

squish after looks

SierraKomodo reviewed (commented):
Nothing strikes me as broken and it compiles. Good for squish.

quardbreak commented on code/controllers/configuration.dm:
`var/static/usealienwhitelistSQL = 0`

Spookerton replied:
tank u, I suppose I should do the default values for completeness too

Spookerton committed:
clean up configuration datum

/datum/configuration -> /configuration
removed legacy initialization pattern
comments -> doc comments
everything is static
gamemode_cache properly lives inside it

Spookerton commented:
comments -> doc comments on defaults + squished

Spookerton commented:
rebased to handle conflict with #30871

Spookerton commented:
Gonna go on ahead and put this in due to the tiny resolution diff
